core/corehttp/redirect.go

```diff
@@ -20,9 +26,14 @@ func RedirectOption(path string, redirect string) ServeOption {
 }
 
 type redirectHandler struct {
-	path string
+	path    string
+	headers map[string][]string
 }
 
 func (i *redirectHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	for k, v := range i.headers {
+		w.Header()[k] = v
```

Comment:
Missed this: this shortcut may cause bugs – we dont normalize header keys to canonical form.

@hacdias   do you mind swapping this to use [`Header.Set`](https://pkg.go.dev/net/http#Header.Set) instead? This will make headers work no matter what notation was used by user.  There is similar snippet in `./core/corehttp/gateway_writable.go` that also needs to be fixed.

Reply:
https://github.com/ipfs/kubo/pull/9734